server/src/main/java/org/opensearch/gateway/remote/model/RemotePinnedTimestamps.java

```diff
@@ -0,0 +1,132 @@
+    private static final Logger logger = LogManager.getLogger(RemotePinnedTimestamps.class);
+
+    public static class PinnedTimestamps extends AbstractDiffable<ClusterBlocks> {
+        private final Map<Long, List<String>> pinnedTimestampPinningEntityMap;
+
+        public PinnedTimestamps(Map<Long, List<String>> pinnedTimestampPinningEntityMap) {
+            this.pinnedTimestampPinningEntityMap = new ConcurrentHashMap<>(pinnedTimestampPinningEntityMap);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeMap(pinnedTimestampPinningEntityMap, StreamOutput::writeLong, StreamOutput::writeStringCollection);
+        }
+
+        public static PinnedTimestamps readFrom(StreamInput in) throws IOException {
+            return new PinnedTimestamps(in.readMap(StreamInput::readLong, StreamInput::readStringList));
+        }
+
+        public void pin(Long timestamp, String pinningEntity) {
+            logger.debug("Pinning timestamp = {} against entity = {}", timestamp, pinningEntity);
+            pinnedTimestampPinningEntityMap.computeIfAbsent(timestamp, k -> new ArrayList<>()).add(pinningEntity);
+        }
+
+        public void unpin(Long timestamp, String pinningEntity) {
+            logger.debug("Unpinning timestamp = {} against entity = {}", timestamp, pinningEntity);
+            pinnedTimestampPinningEntityMap.computeIfPresent(timestamp, (k, v) -> {
+                v.remove(pinningEntity);
+                return v.isEmpty() ? null : v;
+            });
+        }
+
+        public Map<Long, List<String>> getPinnedTimestampPinningEntityMap() {
+            return new HashMap<>(pinnedTimestampPinningEntityMap);
+        }
+    }
+
+    public static final String PINNED_TIMESTAMPS = "pinned_timestamps";
+    public static final ChecksumWritableBlobStoreFormat<PinnedTimestamps> PINNED_TIMESTAMPS_FORMAT = new ChecksumWritableBlobStoreFormat<>(
+        PINNED_TIMESTAMPS,
+        PinnedTimestamps::readFrom
+    );
+
+    private PinnedTimestamps pinnedTimestamps;
+
+    public RemotePinnedTimestamps(String clusterUUID, Compressor compressor, NamedXContentRegistry namedXContentRegistry) {
+        super(clusterUUID, compressor, namedXContentRegistry);
+        pinnedTimestamps = new PinnedTimestamps(new HashMap<>());
+    }
+
+    @Override
+    public BlobPathParameters getBlobPathParameters() {
+        return new BlobPathParameters(List.of(PINNED_TIMESTAMPS), PINNED_TIMESTAMPS);
+    }
+
+    @Override
+    public String getType() {
+        return PINNED_TIMESTAMPS;
+    }
+
+    @Override
+    public String generateBlobFileName() {
+        return this.blobFileName = String.join(DELIMITER, PINNED_TIMESTAMPS, RemoteStoreUtils.invertLong(System.currentTimeMillis()));
+    }
+
+    @Override
+    public ClusterMetadataManifest.UploadedMetadata getUploadedMetadata() {
+        return null;
```

Comment:
curious about what is the use of this .

Reply:
This interface needs to decouple itself from Cluster metadata(introduced in the context of remote cluster state transfers)

Reply:
Refactored : https://github.com/opensearch-project/OpenSearch/pull/15017

Reply:
Refactored : https://github.com/opensearch-project/OpenSearch/pull/15210 Please take a look

Reply:
Thanks @Bukhtawar. I have rebased changes in this PR on main.